File: Placement Sheet/Stack & Queue/ParenthesisChecker.ts
class ParenthesisChecker {
    isMatchingPair(open: string, close: string): boolean {
        return (
            (open === '(' && close === ')') ||
            (open === '{' && close === '}') ||
            (open === '[' && close === ']')
        )
    }
    isBalanced(expression: string): boolean {
        const stack: string[] = [];
        for (const char of expression) {
            if (['(', '{', '['].includes(char)) {
                stack.push(char);
            } else if ([')', '}', ']'].includes(char)) {
                if (stack.length == 0 || !this.isMatchingPair(stack.pop()!, char)) {
                    return false;
                }
            }
        }
        return stack.length === 0;
    }
}

let expression: string = "[{{()}}]"

let checker: ParenthesisChecker = new ParenthesisChecker();

let isBalanced: boolean = checker.isBalanced(expression);

isBalanced == true ? console.log("Expression is balanced...") : console.log("Expression is not balanced...");

export { };

File: Placement Sheet/Stack & Queue/ParenthesisChecker.java
import java.util.Stack;

public class ParenthesisChecker {

    private static boolean isMatchingPair(char open, char close) {
        return ((open == '(' && close == ')') ||
                (open == '{' && close == '}') ||
                (open == '[' && close == ']'));
    }

    public static boolean isBalanced(String expression) {
        Stack<Character> stack = new Stack<>();
        for (char ch : expression.toCharArray()) {
            if (ch == '(' || ch == '{' || ch == '[') {
                stack.push(ch);
            } else if (stack.isEmpty() || !isMatchingPair(stack.pop(), ch)) {
                return false;
            }
        }
        return stack.isEmpty();
    }

    public static void main(String[] args) {
        String expression = "[{({[]})}]";
        boolean isBalanced = isBalanced(expression);
        if (isBalanced) {
            System.out.println("Expression is balanced...");
        } else {
            System.out.println("Expression is not balanced...");
        }
    }
}